### PR TITLE
[16.0][MIG] mis_builder_cash_flow: Migration to version 16.0

### PR DESCRIPTION
**Pull request by stefan-tecnativa**
cc @Tecnativa TT43621

@pedrobaeza @andreampiovesana please review!

This PR is a superseed of #1046

### CODE REVIEW

File: mis_builder_cash_flow/i18n/es.po [hunk 1]
```diff
@@ -6,55 +6,63 @@ msgid ""
...
 "Plural-Forms: nplurals=2; plural=n != 1;\n"
-"X-Generator: Weblate 4.17\n"
+"X-Generator: Weblate 4.3.2\n"
 
 #. module: mis_builder_cash_flow
 #: model:mis.report.instance.period,name:mis_builder_cash_flow.mis_period_plus_1w
+#, fuzzy
 msgid "+1w"
-msgstr "+1s"
+msgstr "+1w"
```
Comment (pedrobaeza) [inline mis_builder_cash_flow/i18n/es.po]: Why changing this?

File: mis_builder_cash_flow/models/mis_report_instance.py [hunk 2]
```diff
@@ -9,9 +9,6 @@ class MisReportInstancePeriod(models.Model):
     def _get_additional_move_line_filter(self):
         """Add the posted condition ."""
         domain = super()._get_additional_move_line_filter()
-        if (
-            self._get_aml_model_name() == "mis.cash_flow"
-            and self.report_instance_id.target_move == "posted"
-        ):
+        if self.report_instance_id.target_move == "posted":
```
Comment (pedrobaeza) [inline mis_builder_cash_flow/models/mis_report_instance.py]: Isn't this done on super already?

File: mis_builder_cash_flow/report/mis_cash_flow_views.xml [hunk 3]
```diff
@@ -102,4 +102,5 @@
             </search>
         </field>
     </record>
+
```
Comment (pedrobaeza) [inline mis_builder_cash_flow/report/mis_cash_flow_views.xml]: Don't add this useless diff

File: mis_builder_cash_flow/views/mis_cash_flow_forecast_line_views.xml [hunk 4]
```diff
@@ -9,7 +9,7 @@
             <form string="MIS Cash Flow Forecast Line">
                 <sheet>
                     <group col="2">
-                        <field name="company_id" groups="base.group_multi_company" />
+                        <field name="company_id" />
```
Comment (pedrobaeza) [inline mis_builder_cash_flow/views/mis_cash_flow_forecast_line_views.xml]: Don't remove this. Instead, add again the same field, but with invisible=1.

**Review by pedrobaeza — COMMENTED** [review]
Co-Author line is not correctly put. You only need to add your name, but as fully qualified with email:

```
Co-Authored-By: Stefan Ungureanu <stefan.ungureanu@tecnativa.com>
```

**Comment by stefan-tecnativa** [comment]
> 

Reviewed all the changes. Thanks for the suggestions. Now I also know how to add correctly the Co-Authored

**Review by pedrobaeza — APPROVED** [review]

**Review by andreampiovesana — CHANGES_REQUESTED** [review]
you should rename account_internal_type (old name) in account_type
and in mis_cash_flow_forecast_line_view_form there are 2 company_id fields

**Review by andreampiovesana — APPROVED** [review]
LGTM

**Review by andreampiovesana — CHANGES_REQUESTED** [review]
<field
                            name="account_id"
                            domain="[('company_id', '=', company_id)]"
                            groups="base.group_multi_company"
                        />
you should remove groups="base.group_multi_company"
immagine
immagine (2)

**Comment by stefan-tecnativa** [comment]
> ```
>                     <field
>                         name="account_id"
>                         domain="[('company_id', '=', company_id)]"
>                         groups="base.group_multi_company"
>                     />
> ```
> 
> you should remove groups="base.group_multi_company" immagine immagine (2)

Thanks for all the suggestions, now it is corrected. It is necessary to add two fields, if not the account_id field returns an error because company_id is not allways in the view.

**Comment by pedrobaeza** [comment]
@stefan-tecnativa what you have to add 2 times is the field `company_id`, not the other, one invisible=1 and without group, and the other visible with the group.

File: mis_builder_cash_flow/report/mis_cash_flow_views.xml [hunk 5]
```diff
@@ -0,0 +1,105 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2019 ADHOC SA
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <record model="ir.ui.view" id="mis_cash_flow_tree_view">
+        <field name="name">mis.cash_flow.tree</field>
+        <field name="model">mis.cash_flow</field>
+        <field name="arch" type="xml">
+            <tree create="false" edit="false" delete="false">
+                <field name="company_id" groups="base.group_multi_company" />
```
Comment (pedrobaeza) [inline mis_builder_cash_flow/report/mis_cash_flow_views.xml:10]: Put as well the field without group as invisible.

File: mis_builder_cash_flow/views/mis_cash_flow_forecast_line_views.xml [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+        <field name="name">mis.cash_flow.forecast_line.tree</field>
+        <field name="model">mis.cash_flow.forecast_line</field>
+        <field name="arch" type="xml">
+            <tree editable="top">
+                <!-- pylint:disable=duplicate-xml-fields -->
+                <field name="company_id" groups="base.group_multi_company" />
+                <field
+                    name="company_id"
+                    invisible="1"
+                    groups="!base.group_multi_company"
```
Comment (pedrobaeza) [inline mis_builder_cash_flow/views/mis_cash_flow_forecast_line_views.xml]: Don't put any group. There's no problem in having several times the same field (it's officially supported), but even this one is invisible.

File: mis_builder_cash_flow/report/mis_cash_flow_views.xml (hunk 5 above)
Comment (pedrobaeza) [reply mis_builder_cash_flow/report/mis_cash_flow_views.xml:10]: Sorry, in this case there's a pre-commit error, so please remove it again.

**Review by ioans73 — APPROVED** [review]
Functional review

File: mis_builder_cash_flow/i18n/es.po [hunk 7]
```diff
@@ -6,53 +6,61 @@ msgid ""
...
 
 #. module: mis_builder_cash_flow
 #: model:mis.report.instance.period,name:mis_builder_cash_flow.mis_period_plus_4w
+#, fuzzy
 msgid "+4w"
 msgstr "+4s"
 
 #. module: mis_builder_cash_flow
 #: model:mis.report.instance.period,name:mis_builder_cash_flow.mis_period_plus_5w
+#, fuzzy
```
Comment (pedrobaeza) [inline mis_builder_cash_flow/i18n/es.po]: Please remove all these fuzzy lines added here

**Review by andreampiovesana — APPROVED** [review]
LGTM

File: mis_builder_cash_flow/i18n/es.po [hunk 8]
```diff
@@ -221,7 +221,7 @@ msgstr "Línea de previsión de flujo de caja MIS"
 #. module: mis_builder_cash_flow
 #: model:ir.model,name:mis_builder_cash_flow.model_mis_report_instance_period
 msgid "MIS Report Instance Period"
-msgstr "Periodo Informe MIS"
+msgstr ""
```
Comment (pedrobaeza) [inline mis_builder_cash_flow/i18n/es.po]: You have removed existing translations.

File: mis_builder_cash_flow/i18n/es.po [hunk 9]
```diff
@@ -252,7 +252,7 @@ msgstr "Previsión Salidas"
 #. module: mis_builder_cash_flow
 #: model:mis.report.kpi,description:mis_builder_cash_flow.mis_kpi_out_payable
 msgid "Out payable"
-msgstr "Cuentas a pagar"
+msgstr "Salidas cuentas por pagar"
```
Comment (pedrobaeza) [inline mis_builder_cash_flow/i18n/es.po]: This new translation is incorrect

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**